framework/resource/retry/resource.go

```diff
@@ -1,4 +1,4 @@
-package framework
+package retry
```

Comment:
In the retry resource changed nothing. Here is only some renaming because the file was moved to its own package. I like it is in its own package now. The drawback was to have a dedicated `spec` package, but I am fine with this.